catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java

```diff
@@ -256,69 +261,72 @@ public ResourceResponse retrieveResource(URI resourceURI, Map<String, Serializab
+        LOGGER.debug("resource name: {}", fileAddress);
+        return retrieveHttpProduct(resourceURI, fileAddress, bytesToSkip, properties);
+      case URL_FILE_SCHEME:
+        LOGGER.debug("Resource URI is a File");
+        File filePathName = new File(resourceURI);
+        if (validateFilePath(filePathName)) {
+          String fileName = filePathName.getName();
+          LOGGER.debug("resource name: {}", fileName);
+          return retrieveFileProduct(resourceURI, fileName, bytesToSkip);
+        } else {
+          throw new ResourceNotFoundException(
+              "Error retrieving resource ["
+                  + resourceURI.toString()
+                  + "]. Invalid Resource URI of ["
+                  + resourceURI.toString()
+                  + "]. Resources must be in one of the following directories: "
+                  + this.rootResourceDirectories.toString());
+        }
+      default:
         throw new ResourceNotFoundException(
-            "Error retrieving resource ["
-                + resourceURI.toString()
-                + "]. Invalid Resource URI of ["
-                + resourceURI.toString()
-                + "]. Resources must be in one of the following directories: "
-                + this.rootResourceDirectories.toString());
-      }
-    } else {
-      ResourceNotFoundException ce =
-          new ResourceNotFoundException(
-              "Resource qualifier ( "
-                  + resourceURI.getScheme()
-                  + " ) not valid. "
-                  + URLResourceReader.TITLE
-                  + " requires a qualifier of "
-                  + URL_HTTP_SCHEME
-                  + " or "
-                  + URL_HTTPS_SCHEME
-                  + " or "
-                  + URL_FILE_SCHEME);
-      throw ce;
+            "Resource qualifier ( "
+                + resourceURI.getScheme()
+                + " ) not valid. "
+                + URLResourceReader.TITLE
+                + " requires a qualifier of "
+                + URL_HTTP_SCHEME
+                + " or "
+                + URL_HTTPS_SCHEME
+                + " or "
+                + URL_FILE_SCHEME);
     }
   }
 
   private ResourceResponse retrieveFileProduct(
       URI resourceURI, String productName, String bytesToSkip) throws ResourceNotFoundException {
-    URLConnection connection = null;
+    URLConnection connection;
     try {
       LOGGER.debug("Opening connection to: {}", resourceURI);
       connection = resourceURI.toURL().openConnection();
 
+      final String originalFileName = productName;
       productName =
-          StringUtils.defaultIfBlank(
-              handleContentDispositionHeader(
-                  connection.getHeaderField(HttpHeaders.CONTENT_DISPOSITION)),
-              productName);
+          AccessController.doPrivileged(
+              (PrivilegedAction<String>)
+                  () ->
+                      StringUtils.defaultIfBlank(
```

Comment:
I'm not entirely sure why this was the correct operation to do a privileged action on, and not when we get the input stream from the connection below in this same method.

Reply:
I'm not sure why you'd need it around this block at all. The tika one makes sense, but not this one.

Reply:
☝️

Reply:
There's a few lines below this where the input stream is read from the connection where I originally thought the problem would be and put the privileged action, but was still running into permission issues. I debugged through and found the issue here. Seems weird to me too.